### PR TITLE
Fix NPE thrown when `<KeyStoreMapping>` Tag Not Available.

### PR DESCRIPTION
**Pull request by Binara-Sachin**
- This PR fixes the NPE thrown when the `<KeyStoreMapping>` tag is not present in the `identity.xml` file.
- Related to https://github.com/wso2/carbon-identity-framework/pull/5794
---
- Problem Statement: wso2-enterprise/iam-product-management#67
- Git Issue: wso2/product-is#20564

### CODE REVIEW

**Review by pamodaaw — APPROVED** [review]

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/IdentityKeyStoreResolver.java [hunk 1]
```diff
@@ -535,6 +535,10 @@ private void parseIdentityKeyStoreMappingConfigs() {
                 IdentityKeyStoreResolverUtil.getQNameWithIdentityNameSpace(
                         IdentityKeyStoreResolverConstants.CONFIG_ELEM_KEYSTORE_MAPPING));
 
+        if (keyStoreMappingsElem == null) {
```
Comment (pamodaaw) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/IdentityKeyStoreResolver.java:538]: Is the above logic is null-safe?

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/IdentityKeyStoreResolver.java [hunk 2]
```diff
@@ -530,10 +530,25 @@ private String getCustomKeyStoreConfig(String keyStoreName, String configName)
...
+                                IdentityKeyStoreResolverConstants.CONFIG_ELEM_KEYSTORE_MAPPING));
+            }
+        }
+
+        if (keyStoreMappingsElem == null) {
+            LOG.warn(String.format("%s.%s element not found in identity.xml file. " +
+                            "You may ne using an outdated or  identity.xml.j2 file.",
+                    IdentityKeyStoreResolverConstants.CONFIG_ELEM_SECURITY,
+                    IdentityKeyStoreResolverConstants.CONFIG_ELEM_KEYSTORE_MAPPING));
+            return;
```
Comment (pamodaaw) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/IdentityKeyStoreResolver.java:550]:
```suggestion
            LOG.warn(String.format("The element %s.%s is not found in identity.xml file. ",                    IdentityKeyStoreResolverConstants.CONFIG_ELEM_SECURITY,
                    IdentityKeyStoreResolverConstants.CONFIG_ELEM_KEYSTORE_MAPPING));
            return;
```

Comment (Binara-Sachin) [reply components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/IdentityKeyStoreResolver.java]: Fixed in https://github.com/wso2/carbon-identity-framework/pull/5978/commits/ef551e1bb2a5d092d74f0b968eb87df4a952ca20

File: components/identity-core/org.wso2.carbon.identity.core/src/test/java/org/wso2/carbon/identity/core/IdentityKeyStoreResolverTest.java [hunk 3]
```diff
@@ -188,6 +161,40 @@ public void testGetInstance() {
...
+                    fileName).toString();
+
+            // Mock IdentityConfigParser.
+            mockIdentityConfigParser = IdentityConfigParser.getInstance(identityXmlPath);
+            identityConfigParser.when(IdentityConfigParser::getInstance).thenReturn(mockIdentityConfigParser);
+
+            // Test instance creation --> Config read.
+            IdentityKeyStoreResolver.getInstance();
+        } catch (Exception e) {
+            fail();
```
Comment (pamodaaw) [inline components/identity-core/org.wso2.carbon.identity.core/src/test/java/org/wso2/carbon/identity/core/IdentityKeyStoreResolverTest.java:194]: Please add the exception details in the fail()
ex:     `fail("Test failed due to exception: " + e.getMessage());`

**Review by pamodaaw — APPROVED** [review]

**Pull request merged**